Tests/UnitTests/Caching/FileRepositoryTests.swift

```diff
@@ -0,0 +1,127 @@
+import XCTest
+
+@MainActor
+class FileRepositoryTests: TestCase {
+    let someURL = URL(string: "https://somesite.com/someurl").unsafelyUnwrapped
+
+    func test_ifContentExists_networkServiceIsNotCalled() async throws {
+        let sut = await makeSUT()
+        sut.cache.stubCachedContentExists(with: true)
+        let data = try await sut.fileRepository.getCachedURL(for: someURL)
+
+        XCTAssertNotNil(data)
+        XCTAssertEqual(sut.networkService.invocations, [])
+    }
+
+    func test_prefetch_invokesNetwork() async throws {
+        let sut = await makeSUT()
+        let data = "SomeData".data(using: .utf8).unsafelyUnwrapped
+
+        sut.cache.stubSaveData(with: .success(.init(data: data, url: someURL)))
+        sut.cache.stubCachedContentExists(with: false)
+        sut.networkService.stubResponse(at: 0, result: .success(data))
+
+        sut.fileRepository.prefetch(urls: [someURL])
+
+        await yield()
+
+        XCTAssertEqual(sut.networkService.invocations.count, 1)
+    }
+
+    func test_whenCacheURLCannotBeAssembled_returnsNil() async throws {
+        let sut = await makeSUT(cacheDirectoryURL: nil)
+        do {
+            _ = try await sut.fileRepository.getCachedURL(for: someURL)
+        } catch {
+            switch error as? FileRepository.Error {
+            case .failedToCreateCacheDirectory: break
+            default:
+                XCTFail(#function)
+            }
+        }
+
+        XCTAssertEqual(sut.cache.cachedContentExistsInvocations, [])
+    }
+
+    func test_whenNetworkServiceFails_completesWithError() async {
+        let sut = await makeSUT()
+
+        sut.cache.stubCachedContentExists(with: false)
+        sut.networkService.stubResponse(at: 0, result: .failure(SampleError()))
+        do {
+            _ = try await sut.fileRepository.getCachedURL(for: someURL)
+            XCTFail(#function)
+        } catch {
+            switch error as? FileRepository.Error {
+            case .failedToFetchFileFromRemoteSource:
+                break
+            default:
+                XCTFail(#function)
+            }
+        }
+    }
+
+    func test_savingData_mapsURLToNewURLType_andReturnsIt() async throws {
+        let sut = await makeSUT()
+        let data = "SomeData".data(using: .utf8).unsafelyUnwrapped
+        sut.cache.stubCachedContentExists(with: false)
+        sut.cache.stubSaveData(with: .success(.init(data: data, url: someURL)))
+        sut.networkService.stubResponse(at: 0, result: .success(data))
+        let result = try await sut.fileRepository.getCachedURL(for: someURL)
+
+        let expectedCachedURL = URL(string: "data:sample/someurl").unsafelyUnwrapped
+
+        XCTAssertEqual(sut.networkService.invocations, [someURL])
+        XCTAssertEqual(sut.cache.saveDataInvocations, [.init(data: data, url: expectedCachedURL)])
+        XCTAssertEqual(result, expectedCachedURL)
+        XCTAssertNotEqual(someURL, expectedCachedURL)
+    }
+
+    func makeSUT(
```

Comment:
```suggestion
    func makeSystemUnderTest(
```

Small nit but just for clarity 😇 Odds are I will never remember what SUT means 🫠